figo/figo.py

```diff
@@ -510,12 +510,10 @@ def remove_account(self, account_or_account_id):
         :Parameters:
          - `account_or_account_id` - account to be removed or its ID
         """
-        if isinstance(account_or_account_id, STRING_TYPES):
-            self._request_with_exception("/rest/accounts/%s" % account_or_account_id,
-                                           method="DELETE")
-        else:
-            self._request_with_exception("/rest/accounts/%s" % account_or_account_id.account_id,
-                                           method="DELETE")
```

Comment:
Smell - yes. Breaking change - obviously, this is why I left it there. Changing `__str__` - sounds like a hack; you might want to use `__str__` for debugging and showing only the id would not be very useful.

In the long run this kind of call should be generated automatically from the API schema anyway. Whether to pass objects or ids, polymorphic or in different functions, should be discussed.